mist-core/src/main/java/edu/snu/mist/core/master/recovery/RecoveryStarter.java

```diff
@@ -60,16 +61,20 @@ public RecoveryStarter(final Map<String, GroupStats> failedGroupMap,
 
   @Override
   public void run() {
-    if (taskRequestor != null) {
-      // Request an evaluator for recovery task if recoveryScheduler is not null.
-      // If recovery scheduler is null, it does not request a new task.
-      taskRequestor.setupTaskAndConn(1);
+    try {
+      if (taskRequestor != null) {
+        // Request an evaluator for recovery task if recoveryScheduler is not null.
+        // If recovery scheduler is null, it does not request a new task.
+        taskRequestor.setupTaskAndConn(1);
+      }
+      // Start recovering of the queries...
+      recoveryScheduler.startRecovery(failedGroupMap);
+      final long startTime = System.currentTimeMillis();
+      // Blocks the thread until the recovery has been finished...
+      recoveryScheduler.awaitUntilRecoveryFinish();
+      LOG.log(Level.INFO, "Recovery is finished in {0} ms...", System.currentTimeMillis() - startTime);
+    } catch (final AvroRemoteException | InterruptedException e) {
```

Comment:
Where does this exception can occur? it would be good to reduce the area of `try{ ...}` clause.